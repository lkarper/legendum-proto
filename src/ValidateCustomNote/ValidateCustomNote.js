import React, { useEffect } from 'react';
import PropTypes from 'prop-types';

const ValidateCustomNote = (props) => {
    const { 
        customNoteError,
        customNote,
        setCustomNoteError,
    } = props;

    useEffect(() => {
        if (customNote.length && !/\w|\d/.test(customNote)) {
            setCustomNoteError('Note must contain at least one letter or number');
        } else if (customNote.length > 1000) {
            setCustomNoteError('Note cannot exceed 1000 characters in length.');
        } else {
            setCustomNoteError('');
        }

    }, [customNote, setCustomNoteError]);

    return (
        <div role="alert">
            {customNoteError}
        </div>
    );
}

ValidateCustomNote.defaultProps = { 
    customNoteError: '',
    customNote: '',
    setCustomNoteError: () => {},
};

ValidateCustomNote.propTypes = { 
    customNoteError: PropTypes.string,
    customNote: PropTypes.string,
    setCustomNoteError: PropTypes.func,
};

export default ValidateCustomNote;
