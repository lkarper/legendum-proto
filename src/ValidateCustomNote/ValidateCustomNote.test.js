import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import ValidateCustomNote from './ValidateCustomNote';

describe('ValidateCustomNote component', () => {
    it('renders the UI without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<ValidateCustomNote />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const wrapper = mount(<ValidateCustomNote />);
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });
});
