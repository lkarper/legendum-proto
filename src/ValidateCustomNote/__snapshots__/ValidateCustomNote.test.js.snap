// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ValidateCustomNote component renders the UI as expected 1`] = `
<ValidateCustomNote
  customNote=""
  customNoteError=""
  setCustomNoteError={[Function]}
>
  <div
    role="alert"
  />
</ValidateCustomNote>
`;
